### PR TITLE
🐛 octavia-cli: specific release job for octavia

### PR DESCRIPTION
**Pull request by alafanechere**
## What
The current release workflow does not work for Octavia because:
* Octavia CLI builds needs Python 3.8
* The current release workflow uses Python 3.7
* The self-hosted runner docker version does not have the `docker buildx` command to build multi-arch images, which is required for Octavia.

### 1. Common `bump_version.sh` script
I move the version bumping logic to a separate script which is re-used by the three jobs: 
* `releaseAirbyte` job, called from `release_version.sh`
* `releaseOctavia` job, called from `release_version_octavia.sh`
* `createPullRequest` job, called from a dedicated step
There is no harm in bumping three times in these three jobs because they all check out the same branch and only the `createPullRequest` job commits the changes. Other jobs require version bumping to tag the images after build.

### 2. Specific Octavia release job
1. Use `ubuntu-latest` runner, on which `docker buildx` is available
2. Install and use python 3.8 on this job
3. Run `release_version_octavia.sh` which is bumping, building and publishing octavia

### 3. Create a final job to open the bump PR
This job runs after `releaseAirbyte` and `releaseOctavia`:
1. Checkout Airbyte repo
2. Bump version
3. Create and open the bump version PR

## It works 🎉 
https://github.com/airbytehq/airbyte/pull/11587/files

The docker images were also successfully pushed to the DockerHub for Octavia, but I deleted the tags.
Did not run the platform build and publish yet, for safety mainly 😄 
## 🚨User impact
The usual release process should not be impacted.

### CODE REVIEW

**Comment by git-phu** [comment]
merged in #11518 to get the release workflow working again on master branch

**Comment by alafanechere** [comment]
I'm attempting a rework as suggested by Davin to make a single bump version PR for both Airbyte and octavia.

File: tools/bin/release_version_octavia.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+. tools/lib/lib.sh
+
+if [[ -z "${DOCKER_PASSWORD}" ]]; then
+  echo 'DOCKER_PASSWORD for airbytebot not set.';
+  exit 1;
+fi
+
+docker login -u airbytebot -p "${DOCKER_PASSWORD}"
```
Comment (supertopher) [inline tools/bin/release_version_octavia.sh:12]: nit: again for noobs and clarity: long flags in shell are intended for scripts and short flags are intended for interactive shells.  to save future users from looking what the short flags are:
```suggestion
docker login --username airbytebot --password "${DOCKER_PASSWORD}"
```

**Review by supertopher — APPROVED** [review]
I made this comments before you changed some stuff so ¯\_(ツ)_/¯ 

but looks great overall

File: tools/bin/bump_version.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -eu
+PREV_VERSION=$(grep -w VERSION .env | cut -d"=" -f2)
+GIT_REVISION=$(git rev-parse HEAD)
+
+pip install bumpversion
+bumpversion "$PART_TO_BUMP"
+
+NEW_VERSION=$(grep -w VERSION .env | cut -d"=" -f2)
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply tools/bin/bump_version.sh:9]: I can retrieve the new version from the `bumpversion` execution, but I wanted to use the same grep that we use for `PREV_VERSION` for consistency. Thanks for the suggestion! Do you think the grep regex should be more strict for safety?

File: tools/bin/release_version.sh [hunk 3]
```diff
@@ -21,24 +21,11 @@ fi
...
-[[ -z "$GIT_REVISION" ]] && echo "Couldn't get the git revision..." && exit 1
-
-echo "Bumped version from ${PREV_VERSION} to ${NEW_VERSION}"
-echo "Building and publishing version $NEW_VERSION for git revision $GIT_REVISION..."
-
-VERSION=$NEW_VERSION SUB_BUILD=PLATFORM ./gradlew clean build
-SUB_BUILD=PLATFORM ./gradlew publish
-VERSION=$NEW_VERSION GIT_REVISION=$GIT_REVISION docker-compose -f docker-compose.build.yaml push
-echo "Completed building and publishing..."
+source ./tools/bin/bump_version.sh
```
[the remark this answers is not in the document]
Comment (supertopher) [reply tools/bin/release_version.sh:24]: Screen Shot 2022-03-30 at 11 59 31 AM

you will get those variables.  However you will also execute all the code.  I made a screenshot example.  Just like most `import` statements you get both the variables and you execute the code

File: tools/bin/bump_version.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (supertopher) [reply tools/bin/bump_version.sh:9]: greps are fine, I've just been bitten in the past.  

They are brittle to future changes and it's not usually clear when a break happens caused by changes to something like ENV variable naming

**Review by supertopher — APPROVED** [review]
still approving

**Pull request merged**